File: app/src/main/java/com/coinninja/coinkeeper/model/db/converter/MemPoolStateConverter.java
package com.coinninja.coinkeeper.model.db.converter;

import com.coinninja.coinkeeper.model.db.enums.MemPoolState;

import org.greenrobot.greendao.converter.PropertyConverter;

public class MemPoolStateConverter implements PropertyConverter<MemPoolState, Integer> {
    @Override
    public MemPoolState convertToEntityProperty(Integer databaseValue) {
        if (databaseValue == null) {
            return null;
        }

        for (MemPoolState state : MemPoolState.values()) {
            if (state.getId() == databaseValue) {
                return state;
            }
        }

        return MemPoolState.PENDING;
    }

    @Override
    public Integer convertToDatabaseValue(MemPoolState state) {
        return state == null ? null : state.getId();
    }
}

File: persistance/src/main/java/app/coinninja/cn/persistance/converter/MemPoolStateConverter.kt
package app.coinninja.cn.persistance.converter

import androidx.room.TypeConverter
import app.coinninja.cn.persistance.model.MemPoolState

object MemPoolStateConverter {
    @TypeConverter
    @JvmStatic
    fun to(id: Int?): MemPoolState? = id?.let { return MemPoolState.from(it) }

    @TypeConverter
    @JvmStatic
    fun from(state: MemPoolState?): Int? = state?.let { return it.id }

}
